File: src/com/nimbusly/isometrix/IntegerMath.java
package com.nimbusly.isometrix;

public class IntegerMath {
	
	public static int round(int value, int divisor) {
		return (value + divisor/2) / divisor;
	}
	
	public static int roundDown(int value, int increment) {
		return floor(value, increment) * increment;
	}
	
	public static int roundUp(int value, int increment) {
		return ceiling(value, increment) * increment;
	}
	
	public static int floor(int value, int divisor) {
		if (value >= 0) {
			return value / divisor;
		} else {
			return -ceiling(-value, divisor);
		}
	}
	
	public static int ceiling(int value, int divisor) {
		if (value >= 0) {
			return (value + divisor - 1) / divisor;
		} else {
			return -floor(-value, divisor);
		}
	}

	public static int signum(int value) {
		return (value > 0) ? +1 : (value == 0) ? 0 : -1;
	}
}

File: src/com/nimbusly/isometrix/IntegerMath.scala
package com.nimbusly.isometrix

object IntegerMath {

	def round(value : Int, divisor : Int) : Int = (value + divisor/2) / divisor
	
	def roundDown(value : Int, increment : Int) : Int = floor(value, increment) * increment
	
	def roundUp(value : Int, increment : Int) : Int = ceiling(value, increment) * increment
	
	def floor(value : Int, divisor : Int) : Int = 
		if (value >= 0) value / divisor else -ceiling(-value, divisor)
	
	def ceiling(value : Int, divisor : Int) : Int = 
		if (value >= 0) (value + divisor - 1) / divisor else -floor(-value, divisor)

	def signum(value : Int) : Int = if (value > 0) +1 else if (value == 0) 0 else -1
}